### PR TITLE
Add registry type and url to worker payload for ECR dispatching

### PR DESCRIPTION
**Pull request by elliott-davis**
tenor-90448144

Closes: #3743

Signed-off-by: Travis Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by elliott-davis** [comment]
I have no way of testing this locally until we get the workers fixed up in local dev

File: components/builder-worker/src/runner/util.rs [hunk 1]
```diff
@@ -196,17 +196,9 @@ pub fn docker_exporter_spec(workspace: &Workspace) -> DockerExporterSpec {
...
-                None
-            } else {
-                Some(val.to_string())
-            }
-        }
-        None => None,
-    };
+    let custom_tag = get_optional_args(&opts, String::from("custom_tag"));
+    let registry_url = get_optional_args(&creds, String::from("registry_url"));
+    let registry_type = get_optional_args(&opts, String::from("integration"));
```
Comment (elliott-davis) [inline components/builder-worker/src/runner/util.rs]: I'm not sure if this needs to come from `opts` or `creds`. You're welcome future me.

File: components/builder-worker/src/runner/util.rs [hunk 2]
```diff
@@ -241,3 +235,18 @@ pub fn docker_exporter_spec(workspace: &Workspace) -> DockerExporterSpec {
         custom_tag: custom_tag,
     }
 }
+
+fn get_optional_args(opts: &JsonValue, arg: String) -> Option<String> {
+    let is_string = format!("{} value is a string", arg);
+    match opts.get(arg) {
+        Some(key) => {
+            let key = key.as_str().expect(&is_string);
```
Comment (christophermaier) [inline components/builder-worker/src/runner/util.rs]: You might want to return a `Result<Option<String>>` here if there's a possibility that the given value isn't a string; calling `expect` will panic.

Alternatively, you might log the error and just return `None` here.

File: components/builder-worker/src/runner/util.rs [hunk 3]
```diff
@@ -202,14 +186,14 @@ pub fn docker_exporter_spec(workspace: &Workspace) -> DockerExporterSpec {
...
-            .get_integrations()
-            .first()
-            .expect("Origin integrations must not be empty")
-            .get_body(),
-    ).expect("Origin integrations body must be JSON");
+    let origin_integration = workspace.job.get_integrations().first().expect(
+        "Origin integrations must not be empty",
+    );
+
+    let creds: JsonValue = serde_json::from_str(origin_integration.get_body()).expect(
```
Comment (chefsalim) [inline components/builder-worker/src/runner/util.rs]: Might as well remove the expect here as well, and explicitly error check (avoid potential future panic)

**Comment by chefsalim** [comment]
tenor-156882762

**Comment by elliott-davis** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/303644428).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/303644428).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**